pandas/tests/io/test_common.py

```diff
@@ -169,6 +175,23 @@ def test_read_fspath_all(self, reader, module, path, datapath):
         mypath = CustomFSPath(path)
         result = reader(mypath)
         expected = reader(path)
+
+        if path.endswith('.pickle'):
+            # categorical
+            tm.assert_categorical_equal(result, expected)
+        else:
+            tm.assert_frame_equal(result, expected)
+
+    def test_read_fspath_all_read_table(self, datapath):
+        pytest.importorskip('os')
```

Comment:
why would os not be available?

Reply:
I would not know… frankly I just went with what was already written in `test_read_fspath_all`, plugging in `tm.assert_produces_warning` where appropriate.

Reply:
Let's remove that line then.